go/vt/servenv/grpc_auth_plugin.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servenv
+
+import (
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+
+	log "github.com/golang/glog"
+)
+
+// AuthPlugin provides an interface to implement auth in Vitess in
+// grpc server
+type AuthPlugin interface {
+	Authenticate(ctx context.Context, fullMethod string) (context.Context, error)
+}
+
+// authPlugins is a registry of AuthPlugin initializers.
+var authPlugins = make(map[string]func() (AuthPlugin, error))
+
+// RegisterAuthPlugin registers an implementation of AuthServer.
+func RegisterAuthPlugin(name string, authPlugin func() (AuthPlugin, error)) {
+	if _, ok := authPlugins[name]; ok {
+		log.Fatalf("AuthPlugin named %v already exists", name)
+	}
+	authPlugins[name] = authPlugin
+}
+
+// GetAuthPlugin returns an AuthPlugin by name, or log.Fatalf.
+func GetAuthPlugin(name string) func() (AuthPlugin, error) {
```

Comment:
`GetAuthenticator`?

Reply:
This can be improved to also invoke the initializer function and just return the an `AuthPlugin` instead.